sdk/python/kfp/cli/pipeline.py

```diff
@@ -42,6 +42,20 @@ def upload(ctx, pipeline_name, package_file):
     _display_pipeline(pipeline)
 
 
+@pipeline.command()
+@click.argument("package-file")
+@click.argument("pipeline-version")
+@click.argument("pipeline-id")
```

Comment:
Would it make sense to have (at least some of them as) options instead? It is error-prone to require a specific order of simple args.

It could be:
```python
@click.option(
    "--id",
    "--pipeline-id",
    help="ID of the pipeline"
)
@click.option(
    "-v",
    "--pipeline-version",
    help="Name of the pipeline version"
)
@click.argument("package-file")
```

Reply:
I was thinking they are the necessary arguments of `client.pipeline_uploads.upload_pipeline_version` when to implement it. I'll make them options. Thanks.

Reply:
@elikatsis I changed them to the options, but I feel they should be arguments because of the following reasons. 

* When to run `client.pipeline_uploads.upload_pipeline_version` without a pipeline id, the API will return a 500 error
* When to run `client.pipeline_uploads.upload_pipeline_version` without a version name, the API will use "None" as the version



Here is the diff to make the arguments options

<details>
<summary>code diff</summary>

```diff
diff --git a/sdk/python/kfp/_client.py b/sdk/python/kfp/_client.py
index 0108314c..34fc25df 100644
--- a/sdk/python/kfp/_client.py
+++ b/sdk/python/kfp/_client.py
@@ -688,18 +688,6 @@ class Client(object):
       IPython.display.display(IPython.display.HTML(html))
     return response
 
-  def upload_pipeline_version(self, pipeline_package_path, version_name, pipeline_id):
-    """Uploads a version of the pipeline to the Kubeflow pipeline cluster.
-    Args:
-      pipeline_package_path: Local path to the pipeline package.
-      version_name: Version name of the pipeline.
-      pipeline_id: The string ID of the pipeline.
-    Returns:
-      A response object including details of a pipeline.
-    """
-    response = self._upload_api.upload_pipeline_version(pipeline_package_path, name=version_name, pipelineid=pipeline_id)
-    return response
-
   def get_pipeline(self, pipeline_id):
     """Get pipeline details.
     Args:
diff --git a/sdk/python/kfp/cli/pipeline.py b/sdk/python/kfp/cli/pipeline.py
index 9abe9a4b..930a5f29 100644
--- a/sdk/python/kfp/cli/pipeline.py
+++ b/sdk/python/kfp/cli/pipeline.py
@@ -43,16 +43,27 @@ def upload(ctx, pipeline_name, package_file):
 
 
 @pipeline.command()
+@click.option(
+    "-p",
+    "--pipeline-id",
+    help="ID of the pipeline"
+)
+@click.option(
+    "-v",
+    "--pipeline-version",
+    help="Name of the pipeline version"
+)
 @click.argument("package-file")
-@click.argument("pipeline-version")
-@click.argument("pipeline-id")
 @click.pass_context
 def upload_version(ctx, package_file, pipeline_version, pipeline_id):
     """Upload a version of the KFP pipeline"""
     client = ctx.obj["client"]
 
-    version = client.upload_pipeline_version(package_file, pipeline_version, pipeline_id)
-    logging.info("The {} version of the pipeline {} has been submitted\n".format(pipeline_version, pipeline_id))
+    version = client.pipeline_uploads.upload_pipeline_version(
+        package_file, name=pipeline_version, pipelineid=pipeline_id)
+    logging.info(
+        "The {} version of the pipeline {} has been submitted\n".format(
+            pipeline_version, pipeline_id))
     _display_pipeline_version(version)

```
</details>

Could you tell me your opinion?

Reply:
I believe you could set the `click.option` argument: `required=True`.
The following should work:
```python
@click.option(
    "-v",
    "--pipeline-version",
    required=True,
    help="Name of the pipeline version"
)
```

Reply:
That makes sense. Thanks!

Reply:
I modified the code in 70d7d44...7ed5434. PTAL.